homeassistant/components/switch/mysensors.py

```diff
@@ -0,0 +1,171 @@
+"""
+homeassistant.components.switch.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for MySensors switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.mysensors.html
+"""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    STATE_ON, STATE_OFF)
+
+import homeassistant.components.mysensors as mysensors
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = ['mysensors']
```

Comment:
I think it is better to leave the dependencies empty and fire discovery events from the main component to set up the switch/sensor platforms when it finds them. That way users will only have to configure the main component in their config.